### ISSUE TITLE
go.mod does not specify v2 version

### ISSUE THREAD

**Issue by hvastani** [shirou/gopsutil#663]
The go.mod file in this repo does not specify the /v2 tag so if this module is imported in any project, it will be imported with +incompatible 
From the official documentation (https://github.com/golang/go/wiki/Modules)

_The module path (such as module foo/v2) declared at the start of a module's go.mod file is both:
the definitive declaration of that module's identity
the definitive declaration of how that module must be imported by consuming code_

_A module can import a v2+ package that has not opted in to modules itself. A non-module v2+ package that has a valid v2+ semver tag will be recorded with an +incompatible suffix in the importing module's go.mod file. The +incompatible suffix indicates that even though the v2+ package has a valid v2+ semver tag such as v2.0.0, the v2+ package has not actively opted in to modules and hence that v2+ package is assumed to have not been created with an understanding of the implications of Semantic Import Versioning and how to use major versions in import paths. Therefore, when operating in module mode, the go tool will treat a non-module v2+ package as an (incompatible) extension of the v1 version series of the package and assume the package has no awareness of Semantic Import Versioning, and the +incompatible suffix is an indication that the go tool is doing so._

Tested this on centos linux and saw this in my go.mod file 
` github.com/shirou/gopsutil v2.18.12+incompatible`
When I tested this, the current tag was 2.19.03
My go version 
`go version go1.11 linux/amd64`

**Comment by Lomanic** [comment]
Looks like we would need a `/v2`  in the import path for it to be not flagged as incompatible (`github.com/shirou/gopsutil/v2/...`), using a `v2` branch it seems according to https://research.swtch.com/vgo-module#from_repository_to_modules

**Comment by shirou** [comment]
I am not familiar with mod yet, but I think go module can adapt [git tag](https://github.com/golang/go/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository) and gopsutil already has version aware git tags. And I don't want to version package directory like `gopsutil/v2/disk/disk_linux.go`

**Comment by ppmathis** [comment]
Please note that aside of breaking Go module standards as mentioned earlier in this issue, this repository also actively breaks the semantic versioning standard due to the patch version containing leading zeroes:

> **Quote from [SemVer](https://semver.org/) paragraph 2:** A normal version number MUST take the form X.Y.Z where X, Y, and Z are non-negative integers, and MUST NOT contain leading zeroes. X is the major version, Y is the minor version, and Z is the patch version. Each element MUST increase numerically. For instance: 1.9.0 -> 1.10.0 -> 1.11.0.

This leads to the site effect of `go get -u github.com/shirou/gopsutil` pulling in **v2.18.12** instead of **v2.19.05**, as it is the newest SemVer-compatible version tag. The newest version of gopsutil has to be pulled using `go get github.com/shirou/gopsutil@2.19.05`, which will end up as a pseudo-version in `go.mod`: `v0.0.0-20180801053943-8048a2e9c577`

Aside from highly advising you to actually pay attention to the Go modules specification and fixing the incorrect import path, I would at least either re-release the current version tag as **v2.19.5** or create a new release as **v2.19.6** to ensure users are working with the latest release.

**Comment by shirou** [comment]
I'm not sure why the go mod thinks v.2.18.12 is newer than v2.19.05. Those are SemVer.

**Comment by ppmathis** [comment]
> I'm not sure why the go mod thinks v.2.18.12 is newer than v2.19.05. Those are SemVer.

Please re-read the paragraph I have quoted from the SemVer specifications page, which states `[...] are non-negative integers, and MUST NOT contain leading zeroes. [...]`. Your currently used version number is violating the SemVer specification as the patch version, which is **5**, is being written with a leading zero as **05**.

**Comment by shirou** [comment]
ah, ok, understood.

So, I tagged `v2.19.6` now. Thank you for your comment!

**Comment by calvn** [comment]
It seems that despite removing the leading zero, `go mod` still pulls it using the pseudo-version:
```
› go get github.com/shirou/gopsutil@v2.19.7
go: finding github.com/shirou/gopsutil v2.19.7
go: downloading github.com/shirou/gopsutil v0.0.0-20190731134726-d80c43f9c984
go: extracting github.com/shirou/gopsutil v0.0.0-20190731134726-d80c43f9c984
```

Pulling `latest` also pins it to `v2.18.12` despite having newer releases that are SemVer compatible:
```
› go get github.com/shirou/gopsutil@latest 
```

go.mod:
```
	...
	github.com/shirou/gopsutil v2.18.12+incompatible // indirect
	...
```

**Comment by Xuanwo** [comment]
@shirou @calvn 

Because there is no `v2` in the import path but the tag is `v2.19.7`, so go mod will use pseudo-version to mark it's a incompatible mod: **treat `v2.19.7` as a simple string instead of a semver tag**.

And pull from latest will get `v2.18.12` because `v2.18.12` is the last version before adapting go module.

So, there are two options here:

- Do not provide a go.mod file, go will always treat this repo in `/v1` but with a high tag.
- Add `/v2` into the path to meet go module requirement.

**Comment by shirou** [comment]
I really don't want to add `/v2` path. So just remove `go.mod` file?

**Comment by Lomanic** [comment]
Adding a `v2` branch might do it, or maybe even default branch could be `v2` instead of `master` (see my first comment and link). But if it means people then would have to `import "github.com/shirou/gopsutil/v2/process" ` I'm also not fond of this. Removing the `go.mod` is simpler.

**Comment by shirou** [comment]
I did those combinations

1. remove `go.mod`
2. changed `go.mod` to add v2 in module  `module github.com/shirou/gopsutil` -> `module github.com/shirou/gopsutil/v2`
3. we still have 0 padding git branch like `v2.19.05` in repository. so rename those.

on my *local pc*. but still have `incompatible` or occurd this error

```
go: finding github.com/shirou/gopsutil/v2 v2.19.8
build github.com/shirou/test: cannot load github.com/shirou/gopsutil/v2/process: module github.com/shirou/gopsutil@latest (v2.18.12+incompatible) found, but does not contain package github.com/shirou/gopsutil/v2/process
```

I have tested on just my local pc, so it might be not affected, but those seem no luck.

**Comment by shirou** [comment]
I am using `go version go1.13 linux/amd64`. Possibly, it time to start v3 ???

**Comment by Xuanwo** [comment]
`v2` should be added to every import of `github.com/shirou/gopsutil`.

`github.com/shirou/gopsutil/v2` and `github.com/shirou/gopsutil` are different import path for go modules.

**Comment by shirou** [comment]
Sorry for the very long time wait. I have released v3 by #938 which has go.mod.
